internal/boxcli/root.go

```diff
@@ -38,20 +36,14 @@ func RootCmd() *cobra.Command {
 			if flags.quiet {
 				cmd.SetErr(io.Discard)
 			}
-
-			// Skip CheckVersion for `devbox global shellenv` because users will include that
-			// command in their shellrc files, and we don't want to bother them everytime they
-			// open their terminals.
-			//
-			// Skip CheckVersion for engineers building devbox during development.
-			if !strings.HasPrefix(cmd.CommandPath(), "devbox global shellenv") &&
-				!build.IsDev {
-				vercheck.CheckVersion(cmd.ErrOrStderr())
-			}
+			vercheck.CheckVersion(cmd.ErrOrStderr(), cmd.CommandPath())
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
 			return cmd.Help()
 		},
+		PersistentPostRun: func(cmd *cobra.Command, args []string) {
+			vercheck.ClearCheckEnvVar()
+		},
```

Comment:
This is not needed. Environment variables set during the execution do not leak to the environment. (in fact that is no way to update the current shell environment which is why the user needs to source shellenv)

Reply:
oh good point